src/policies/networking.ts

```diff
@@ -0,0 +1,97 @@
+import { a } from "pepr";
+
+import { When, containers } from "./common";
+import { exemptHostNamespaces } from "./exemptions/networking";
+
+/**
+ * This policy prevents pods from sharing the host namespaces.
+ *
+ * Host namespaces (Process ID namespace, Inter-Process Communication namespace, and network namespace)
+ * allow access to shared information and can be used to elevate privileges. Pods should not be allowed
+ * access to host namespaces. This policy ensures fields which make use of these host namespaces are
+ * set to `false`.
+ * 
+ * @related https://repo1.dso.mil/big-bang/product/packages/kyverno-policies/-/blob/main/chart/templates/disallow-host-namespaces.yaml
+ */
+When(a.Pod)
+  .IsCreatedOrUpdated()
+  .Validate(request => {
+    if (exemptHostNamespaces(request)) {
+      return request.Approve();
+    }
+
+    const pod = request.Raw.spec!;
+
+    // If the pod is using the host network, IPC, or PID namespaces, deny the request.
+    if (pod.hostNetwork || pod.hostIPC || pod.hostPID) {
+      return request.Deny(
+        "Sharing the host namespaces is disallowed. The fields spec.hostNetwork, spec.hostIPC, and spec.hostPID must not be set to true.",
+      );
+    }
+
+    return request.Approve();
+  });
+
+/**
+ * This policy restricts the use of host ports in Pods.
+ *
+ * Access to host ports can allow potential snooping of network traffic and should be
+ * restricted to a known list. This policy ensures only approved ports
+ * are defined in container's `hostPort` field.
+ * 
+ * @related https://repo1.dso.mil/big-bang/product/packages/kyverno-policies/-/blob/main/chart/templates/restrict-host-ports.yaml
+ */
+When(a.Pod)
+  .IsCreatedOrUpdated()
+  .Validate(request => {
+    // Check all containers in the pod spec, and find the first one that has a host port, if any
+    const hasHostPort = containers(request)
+      .flatMap(c => c.ports || [])
+      .find(p => p.hostPort);
+
+    // If the container has a host port, deny the request
+    if (hasHostPort) {
+      return request.Deny(
+        `Host ports are not allowed. Only exempted resources are allowed to use host ports.`,
```

Comment:
nit: we don't mention exempted resources in all the deny messages, would be good to standardize on whether we add that phrase or not (technically there are no exemptions for this one currently either).

Reply:
```suggestion
        `Host ports are not allowed.`,
```
Don't know if we want to add any "debug" to this one, i.e. `Found: {"name":"my-container","ports":[{hostPort: 80}]}` but at least can drop the reference to exemptions here?